cmd/osm-bootstrap/osm-bootstrap_test.go

```diff
@@ -372,32 +404,60 @@ func TestCreateMeshRootCertificate(t *testing.T) {
+		kubeClient                       kubernetes.Interface
+		configClient                     configClientset.Interface
+		expectDefaultMeshRootCertificate bool
+		expectErr                        bool
 	}{
 		{
-			name:         "MeshRootCertificate found",
-			namespace:    testNamespace,
-			kubeClient:   fakeKube.NewSimpleClientset(),
-			configClient: fakeConfig.NewSimpleClientset([]runtime.Object{testMeshRootCertificate}...),
-			expectErr:    false,
+			name:                             "active default MeshRootCertificate found",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset(),
+			configClient:                     fakeConfig.NewSimpleClientset([]runtime.Object{testActiveDefaultMeshRootCertificate}...),
+			expectDefaultMeshRootCertificate: true,
+			expectErr:                        false,
 		},
 		{
-			name:         "MeshRootCertificate not found but successfully created",
-			namespace:    testNamespace,
-			kubeClient:   fakeKube.NewSimpleClientset([]runtime.Object{testPresetMeshRootCertificate}...),
-			configClient: fakeConfig.NewSimpleClientset(),
-			expectErr:    false,
+			name:                             "active non default MeshRootCertificate found",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset(),
+			configClient:                     fakeConfig.NewSimpleClientset([]runtime.Object{testActiveNonDefaultMeshRootCertificate}...),
+			expectDefaultMeshRootCertificate: false,
+			expectErr:                        false,
 		},
 		{
-			name:         "MeshRootCertificate not found and error creating it",
-			namespace:    testNamespace,
-			kubeClient:   fakeKube.NewSimpleClientset(),
-			configClient: fakeConfig.NewSimpleClientset(),
-			expectErr:    true,
+			name:                             "MeshRootCertificate not found but successfully created",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset([]runtime.Object{testPresetMeshRootCertificate}...),
+			configClient:                     fakeConfig.NewSimpleClientset(),
+			expectDefaultMeshRootCertificate: true,
+			expectErr:                        false,
+		},
+		{
+			name:                             "active MeshRootCertificate not found (passive non default MRC exists) but successfully created",
+			namespace:                        testNamespace,
+			kubeClient:                       fakeKube.NewSimpleClientset([]runtime.Object{testPresetMeshRootCertificate}...),
+			configClient:                     fakeConfig.NewSimpleClientset([]runtime.Object{testPassiveNonDefaultMeshRootCertificate}...),
+			expectDefaultMeshRootCertificate: true,
+			expectErr:                        false,
+		},
+		{
+			name:                             "active MeshRootCertificate not found (passive default MRC exists) and no active MeshRootCertificate created",
```

Comment:
agree deleting a MRC is not something the bootstrapper should do.  This would be a failure state for the mesh in general and would be a scenario created by user-initiated action. I don't think we can do anything here as we don't know the user intent.  Logging a warning as you've done seem appropriate.